### PR TITLE
OSDOCS-1993 Providing a private hosted zone on AWS

### PR DESCRIPTION
**Pull request by codyhoag**
https://issues.redhat.com/browse/OSDOCS-1993

Preview: https://deploy-preview-31315--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-vpc.html

### CODE REVIEW

**Comment by codyhoag** [comment]
@staebler please review and let me know if I left out any details. Thanks!

File: modules/installation-configuration-parameters.adoc [hunk 1]
```diff
@@ -388,6 +388,10 @@ control plane machine pool.
 belong to the same region as the cluster. This is required for regions that require a custom {op-system} AMI.
 |Any published or custom {op-system} AMI that belongs to the set AWS region.
 
+|`platform.aws.hostedZone`
+|An existing Route 53 private hosted zone for the cluster. Providing an existing hosted zone allows you to manually create DNS records for your cluster. You can only use a pre-existing hosted zone when also supplying your own VPC. The hosted zone must already be associated with the user-provided VPC before installation. If undefined, the installation program creates a new hosted zone.
```
Comment (staebler) [inline modules/installation-configuration-parameters.adoc:392]: I am not sure about the following sentence. Even when supplying a hosted zone, the installer and in-cluster operators will still create DNS records in the hosted zone.
> Providing an existing hosted zone allows you to manually create DNS records for your cluster.

Comment (codyhoag) [reply modules/installation-configuration-parameters.adoc:392]: Makes sense; I'll remove that sentence :+1:

**Comment by codyhoag** [comment]
@yunjiang29 PTAL

File: modules/installation-aws-config-yaml.adoc [hunk 2]
```diff
@@ -215,13 +216,14 @@ must belong to the same region as the cluster.
 <9> The AWS service endpoints. Custom endpoints are required when installing to
 an unknown AWS region. The endpoint URL must use the `https` protocol and the
 host must trust the certificate.
+<10> The name of your existing Route 53 private hosted zone. Providing an existing hosted zone requires that you supply your own VPC and the hosted zone is already associated with the VPC prior to installing your cluster. If undefined, the installation program creates a new hosted zone.
```
Comment (yunjiang29) [inline modules/installation-aws-config-yaml.adoc:219]: `The name` -> `The ID` ?

File: modules/installation-configuration-parameters.adoc [hunk 3]
```diff
@@ -388,6 +388,10 @@ control plane machine pool.
 belong to the same region as the cluster. This is required for regions that require a custom {op-system} AMI.
 |Any published or custom {op-system} AMI that belongs to the set AWS region.
 
+|`platform.aws.hostedZone`
+|An existing Route 53 private hosted zone for the cluster. You can only use a pre-existing hosted zone when also supplying your own VPC. The hosted zone must already be associated with the user-provided VPC before installation. If undefined, the installation program creates a new hosted zone.
+|String, for example `Z3URY6TWQ91KVV`.
```
Comment (yunjiang29) [inline modules/installation-configuration-parameters.adoc:393]: @staebler , `hostedZone` ID should correspond to `baseDomain` name， right?

Comment (codyhoag) [reply modules/installation-configuration-parameters.adoc:393]: @staebler @yunjiang29 from the [AWS docs](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/hosted-zone-private-creating.html), it sounds like the domain name is inputted when creating the private hosted zone. Should we warn that the private hosted zone must be associated with the cluster base domain AND user-provided VPC?

Comment (yunjiang29) [reply modules/installation-configuration-parameters.adoc:393]: @codyhoag good suggestion

fyi, @staebler has wrote some restrictions in [PR](https://github.com/openshift/installer/pull/4772#issue-596174283 )

**Review by staebler — COMMENTED** [review]
LGTM

**Comment by codyhoag** [comment]
@yunjiang29 can you confirm? Thanks!

**Comment by yunjiang29** [comment]
@codyhoag LGTM

**Review by adellape — APPROVED** [review]
:+1:

**Pull request merged**